lib/charms/mongodb/v0/helpers.py

```diff
@@ -80,10 +81,37 @@ def get_create_user_cmd(
     ]
 
 
+def get_mongos_args(config: MongoDBConfiguration) -> str:
+    """Returns the arguments used for starting mongos on a config-server side application.
+
+    Returns:
+        A string representing the arguments to be passed to mongos.
+    """
+    # mongos running on the config server communicates through localhost
+    config_server_uri = f"{config.replset}/localhost"
+
+    # no need to add TLS since no network calls are used, since mongos is configured to listen
+    # on local host
+    cmd = [
+        # mongos on config server side should run on 0.0.0.0 so it can be accessed by other units
+        # in the sharded cluster
+        "--bind_ip 0.0.0.0",
```

Comment:
lets enable IPv6 too
```suggestion
        "--bind_ip_all",
```